pkg/interface/src/views/apps/chat/components/lib/chat-window.tsx

```diff
@@ -1,25 +1,64 @@
 import React, { Component, Fragment } from "react";
+import { Virtuoso as VirtualList } from 'react-virtuoso';
 
 import { ChatMessage } from './chat-message';
-import { ChatScrollContainer } from "./chat-scroll-container";
 import { UnreadNotice } from "./unread-notice";
 import { ResubscribeElement } from "./resubscribe-element";
 import { BacklogElement } from "./backlog-element";
+import { Envelope, IMessage } from "~/types/chat-update";
+import { RouteComponentProps } from "react-router-dom";
+import { Patp, Path } from "~/types/noun";
+import { Contacts } from "~/types/contact-update";
+import { Association } from "~/types/metadata-update";
+import { Group } from "~/types/group-update";
+import GlobalApi from "~/logic/api/global";
+
+// This component gets passed an envelopes prop with 20 envelopes
+// Why? No one knows.
+const INITIAL_LOAD = 20; 
+
 
 const MAX_BACKLOG_SIZE = 1000;
 const DEFAULT_BACKLOG_SIZE = 200;
 const PAGE_SIZE = 50;
-const INITIAL_LOAD = 20;
 
 
-export class ChatWindow extends Component {
+type ChatWindowProps = RouteComponentProps<{
+  ship: Patp;
+  station: string;
+}> & {
+  unreadCount: number;
+  envelopes: Envelope[];
+  isChatMissing: boolean;
+  isChatLoading: boolean;
+  isChatUnsynced: boolean;
+  unreadMsg: Envelope | false;
+  stationPendingMessages: IMessage[];
+  mailboxSize: number;
+  contacts: Contacts;
+  association: Association;
+  group: Group;
+  ship: Patp;
+  station: any;
+  api: GlobalApi;
+}
+
+interface ChatWindowState {
+  numPages: number;
+}
+
+export class ChatWindow extends Component<ChatWindowProps, ChatWindowState> {
+  private fetchPending: boolean;
```

Comment:
if you're declaring these members, you can also initialise them here and save you from doing it in the constructor. (It will also infer the type)

Reply:
Sorry, which are you referring to? Point me in the direction of a component with similar syntax if you don't mind.

Reply:
`private fetchPending = false;`